integration-tests/src/test/java/eu/tsystems/mms/tic/testframework/test/webdrivermanager/WebDriverManagerTest.java

```diff
@@ -87,31 +93,54 @@ public void testT02_introduceOwnDriver() {
 
         // assert that webdrivermanager does know about the session.
-        Assert.assertTrue(WebDriverManager.hasSessionsActiveInThisThread(), "WebDriver Session active in this thread.");
+        Assert.assertEquals(WebDriverSessionsManager.getWebDriversFromCurrentThread().count(), 1, "WebDriver Session should active in this thread.");
+        Assert.assertTrue(executionContextController.getCurrentSessionContext().isPresent(), "There should exist a current session context.");
     }
 
     @Test
     public void testT03_MakeSessionExclusive() {
-        WebDriver exclusiveDriver = WebDriverManager.getWebDriver();
-        Assert.assertEquals(WebDriverSessionsManager.getWebDriversFromCurrentThread().count(), 1);
+        WebDriver exclusiveDriver = WEB_DRIVER_MANAGER.getWebDriver();
+        Assert.assertTrue(executionContextController.getCurrentSessionContext().isPresent(), "There should exist a current session context.");
 
-        String sessionId = WebDriverManager.makeSessionExclusive(exclusiveDriver);
+        String sessionId = WEB_DRIVER_MANAGER.makeExclusive(exclusiveDriver);
 
-        Assert.assertEquals(WebDriverSessionsManager.getWebDriversFromCurrentThread().count(), 0);
+        Assert.assertTrue(executionContextController.getCurrentSessionContext().isEmpty(), "The current session context should empty");
 
         Assert.assertNotNull(WebDriverSessionsManager.getSessionContext(exclusiveDriver).get());
 
-        WebDriver driver2 = WebDriverManager.getWebDriver("Session2");
-        WebDriver exclusiveDriverActual = WebDriverManager.getWebDriver(sessionId);
+        WebDriver driver2 = WEB_DRIVER_MANAGER.getWebDriver("Session2");
+        WebDriver exclusiveDriverActual = WEB_DRIVER_MANAGER.getWebDriver(sessionId);
 
         Assert.assertEquals(exclusiveDriver, exclusiveDriverActual, "Got the same WebDriver!");
-        WebDriverManager.shutdownExclusiveSession(sessionId);
+        WEB_DRIVER_MANAGER.shutdownSession(sessionId);
+    }
+
+    private String t03aSessionId = "";
+
+    @Test
+    public void testT03a_SessionExclusivePrepareTest() {
+        WebDriver exclusiveDriver = WEB_DRIVER_MANAGER.getWebDriver();
+        t03aSessionId = WEB_DRIVER_MANAGER.makeExclusive(exclusiveDriver);
+    }
+
+    @Test(dependsOnMethods = "testT03a_SessionExclusivePrepareTest")
+    public void testT03b_SessionExclusiveCheckLinkedSession() {
+        Assert.assertEquals(this.readSessionContextFromMethodContext().count(), 0, "No session should link to current method context");
+
+        WEB_DRIVER_MANAGER.getWebDriver(t03aSessionId);
+
+        Assert.assertEquals(this.readSessionContextFromMethodContext().count(), 1, "A session should link to current method context");
+        this.readSessionContextFromMethodContext()
+                .filter(sessionContext -> sessionContext.getSessionKey().equals(t03aSessionId))
+                .findAny()
+                .ifPresentOrElse(sessionContext -> Assert.assertTrue(true),
```

Comment:
Had to take a double take on this one.
```java
if (this.readSessionContextFromMethodContext().noneMatch(sessionContext -> sessionContext.getSessionKey().equals(t03aSessionId))) {
  Assert.fail("Exclusive session not found in current method context");
}
```
so we do not claim to "assert" something in either case?

(Or prettier if more verbose:
```java
if (this.readSessionContextFromMethodContext()
    .map(SessionContext::getSessionKey)
    .noneMatch(t03aSessionId::equals)) {
  Assert.fail("Exclusive session not found in current method context");
}
```
)

Reply:
Maybe also insert `t03aSessionId` into the assertion message for convenience?
`Assert.fail(String.format("Exclusive session '%s' not found in current method context", t03aSessionId));`